Automattic/wp-calypso
# Full Site Editing: load default template data on plugin activation

#### Changes proposed in this Pull Request

Prerequisite for: https://github.com/Automattic/wp-calypso/pull/33808
Fixes: https://github.com/Automattic/wp-calypso/issues/33636

Preload default template data including page template, header, and footer on plugin activation.
The rendering logic has been modified to fall back to this global default template if specific page
template hasn't been assigned to page's meta field (`_wp_template_id`).

Note: populating data on Dotcom will be handled separately during site creation, and we are tracking that task here https://github.com/Automattic/wp-calypso/issues/33515 (`register_activation_hook` won't work in Dotcom environment).

#### Testing instructions

1. Start with a fresh WP install with Twenty Nineteen theme.
2. Activate the plugin.
3. Navigate to `Templates` CPT and verify that `Page Template` post is created.

<img width="511" alt="Screenshot 2019-06-12 at 17 46 11" src="https://user-images.githubusercontent.com/1182160/59366265-5bd7ce00-8d3a-11e9-9043-8fbcc6e93900.png">

4. It should have `page_template` type assigned for `wp_template_type` taxonomy.
5. Navigate to `Template Parts` CPT and verify that `Header` and `Footer` posts are created.

<img width="516" alt="Screenshot 2019-06-12 at 17 46 25" src="https://user-images.githubusercontent.com/1182160/59366310-701bcb00-8d3a-11e9-8714-0d4e6cc47487.png">

6. They should have `header` and `footer` type assigned to `wp_template_part_type` taxonomy respectively.
7. Create a test page and publish it.
8. Open the page created in step 5 and verify that the default header and footer are replaced.
9. Try to assign a custom page template to your test page, and verify that it overrides the default template.
